api/src/main/java/org/pmiops/workbench/monitoring/GaugeDataCollector.java

```diff
@@ -1,8 +1,27 @@
 package org.pmiops.workbench.monitoring;
 
-import java.util.Map;
-import org.pmiops.workbench.monitoring.views.OpenCensusStatsViewInfo;
+import java.util.Collection;
 
+/**
+ * Each class wishing to supply gauge data to the gauge cron job can simply implement this
+ * interface. By returning a well-formed MeasurementBundle, including any appropriate attachments,
+ * in getGaugeData(), the class is ensured that these values are polled as appropriate.
+ *
+ * <p>Using this arrangement minimizes the added dependencies for instrumented classes. In
+ * particular, they do not depend on the MonitoringService.
+ */
 public interface GaugeDataCollector {
-  Map<OpenCensusStatsViewInfo, Number> getGaugeData();
+
+  /**
+   * Return a collection of @link {MeasurementBundle} objects. Each bundle can have one or more
+   * View/Value pairs and zero or more metadata AttachmentKey/String pairs. There a 1:1
+   * correspondents between MeasurementBundles returned and instantiation of MeasureMaps (and calls
+   * to record).
+   *
+   * <p>If a class has no attachments to provide, then it's fine to include all samples in a single
```

Comment:
good catch. Do we have a way to render our Javadoc currently?

Reply:
I don't think so. I also realized in my own PRs that spotlessApply auto-creates these not-closed paragraph tags, so I don't think it will be super easy to change.